dev/NumberBox/NumberBox.cpp

```diff
@@ -195,9 +193,19 @@ void NumberBox::ValidateInput()
         {
             // Setting NumberFormatter to something that isn't an INumberParser will throw an exception, so this should be safe
             const auto numberParser = NumberFormatter().as<winrt::INumberParser>();
-            const auto parsedNum = numberParser.ParseDouble(text);
 
-            if (!parsedNum)
+            winrt::IReference<double> value;
+
+            if (AcceptsCalculation())
+            {
+                value = NumberBoxParser::Compute(textBox.Text(), numberParser);
+            }
+            else
+            {
+                value = numberParser.ParseDouble(text);
+            }
```

Comment:
```suggestion
auto const value = [this, numberParser]() {
    const auto numberParser = NumberFormatter().as<winrt::INumberParser>();
    if(AcceptsCalculation())
    {
        return NumberBoxParser::Comput(textBox.Text(), numberParser);
    }
    return numberParser.Parse
}();
```

Reply:
I see what you're saying but I just don't personally prefer that, the scope of numberParser isn't that big.

Reply:
It is not just about scope guarding the number parser, it also eliminates the need for the scratch variable value and allows for the local variable value to be defined as const which allows  you to follow CPP guideline CON.4. The CPP guildlines ES.28 specifies the lambda construction as correct.

---
In reply to: [345486058](https://github.com/microsoft/microsoft-ui-xaml/pull/1602#discussion_r345486058) [](ancestors = 345486058)

Reply:
OK I modified it to something we both can hopefully agree on.

Reply:
I like this too :), thanks